virtual-programs/receipt_printer.folk

```diff
@@ -0,0 +1,76 @@
+namespace eval EscPos {
+    proc printProgram {printer id code} {
+        writeFolkFile $id $code
+        writeGeomFile $printer $id
+
+        set matches [Statements::findMatches "/someone/ claims $printer is at /address/"]
+        set matchDict [lindex $matches 0]
+        if {![dict exists $matchDict address]} { return }
+        set address [dict get $matchDict address]
+
+        set printerSocket [socket $address 9100]
+
+        fconfigure $printerSocket -translation binary -buffering none
+        puts -nonewline $printerSocket "[init][tag $id][feed 1]$id ([clock format [clock seconds] -timezone :America/Denver -format "%a, %d %b %Y, %r"])[feed 2]$code[feed 3][cut]"
```

Comment:
I'd love to split this template over multiple lines, but I don't know a good way of doing that in Tcl.

Reply:
Maybe make a list and join by ""?